### PR TITLE
Add milli- prefix to Angstrom unit

### LINKED ISSUE #11745: Add milli- prefix to Angstrom unit
### Description

Is there a particular reason that SI prefixes are disabled for the Angstrom unit?
I am working on a project that prefers to describe the dispersion of a spectrograph in milliangstroms per pixel, and not having this unit is becoming tedious.
I realize the Angstrom is not an SI unit, but I think there is an argument to be made to add at least the milli- prefix.

### PR DESCRIPTION
**Pull request by byrdie**
### Description

Added milli- prefix to Angstrom unit

Fixes #11745

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
Thanks, looks all good. I'll merge once all the tests have passed.

**Comment by pllim** [comment]
Shouldn't there be a test too?

**Comment by mhvk** [comment]
The automatic going through all units should be good enough. But I note there is a failure in the doctests, plus a real error because `mÅ` was used in a test to check that it failed.

@byrdie - do you think you'll be able to fix this? If you feel it is too tricky, I can do it for you.

**Comment by byrdie** [comment]
I will give it a try and let you know if I have any issues.

**Comment by dhomeier** [comment]
This works in some unexpected way in that it enables `u.mAngstrom`, `u.milliAA` and `u.milliangstrom` rather than `u.mAA` and `u.milliAngstrom` - the reason being evidently that for `def_unit`
> the first element is the canonical (short) name, and the rest of the elements are aliases.

It might seem more logical then to define it as
```python
def_unit(['AA', 'Angstrom', 'angstrom'], ...
```
but that would break e.g. the expected fits format outputs.
As a kind of alternative workaround you might instead define both prefix versions as aliases for both the short and long form:
```python
    prefixes=[(['m', 'milli'], ['milli', 'm'], 1.e-3)],
```
You will also have to update the matches for test_units:test_suggestions to
```python
    ('ångström', 'Angstrom, angstrom, mAngstrom or mangstrom')
```

**Comment by byrdie** [comment]
@dhomeier, thanks for explaining that, I had not yet figured out why that test was failing.
I'll add the workaround you posted and update the other test you mentioned.

File: astropy/units/tests/test_units.py [hunk 1]
```diff
@@ -639,7 +639,7 @@ def test_suggestions():
             ('metre', 'meter'),
             ('angstroms', 'Angstrom or angstrom'),
             ('milimeter', 'millimeter'),
-            ('ångström', 'Angstrom or angstrom'),
+            ('ångström', 'Angstrom, angstrom, mAngstrom or mangstrom')
```
Comment (dhomeier) [inline astropy/units/tests/test_units.py:642]:
```suggestion
            ('ångström', 'Angstrom, angstrom, mAngstrom or mangstrom'),
```
lost the comma here

**Review by dhomeier — CHANGES_REQUESTED** [review]
If tests are passing after the fix, this is good to go, but I'd let @mhvk have a second look if he agrees with the multiple alias solution.

**Review by mhvk — APPROVED** [review]
@byrdie - there was still a failure in a doctest (in which `milliangstrom` is now offered - correctly - as an alternative). I went ahead and pushed a change for that to your branch, also squashing your comments in the process. I hope that with this all the tests will pass, because the rest looks all OK. So, I'll approve and merge if tests indeed pass.

Thanks!

**Comment by dhomeier** [comment]
@mhvk this does raise some concerns that similar problems might pop up downstream if looking for the expected exception. But I don't really see a good alternative if we want to accept `u.mAA` (and I definitely think we should).
One option could be to allow only 'milli' as 'short' prefix, backwards as it may seem
`prefixes=[(['milli'], ['milli', 'm'], 1.e-3)],`
but that would still only get rid of `mAngstrom`, not `mangstrom`, so maybe not much point in it.

**Review by dhomeier — APPROVED** [review]

**Comment by byrdie** [comment]
@mhvk, thanks for updating the doctest. I didn't know if I was supposed to update the .rst file directly or if there was a docstring in 
 a .py file somewhere that I had to fix.

**Comment by mhvk** [comment]
@byrdie - yes, I thought this was getting a bit beyond what a supposedly quick PR should do, especially as with those doctests it is often a bit trickier than thought. Anyway, tests pass now and @dhomeier also agrees this is all OK, so I'll merge. Thanks again, and of course other suggestions for making astropy just a little bit better are very welcome!

**Comment by mhvk** [comment]
p.s. @dhomeier - I agree that the `mAngstrom` is really not all that great, but ended up also agreeing that if `mangstrom` is in there, we should just not perfect be the enemy of good enough. It would have been nicer if the names could just have been swapped, so that `AA` was first.

**Comment by dhomeier** [comment]
Thanks @byrdie and congrats to completing the first PR into Astropy!

**Comment by byrdie** [comment]
Thanks @mhvk, @dhomeier and everyone else for your help and allowing me the opportunity to contribute!

**Pull request merged**